packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -366,7 +371,74 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
       });
 
       describe('HBAR Rate Limit For Different Spending Plan Tiers', () => {
+        const createAliasAndAssociateSpendingPlan = async (subscriptionTier: SubscriptionTier) => {
+          const aliasAccount = await Utils.createAliasAccount(
+            mirrorNode,
+            global.accounts[0],
+            requestId,
+            initialBalance,
+          );
+          global.accounts.push(aliasAccount);
+
+          const hbarSpendingPlan = await hbarSpendingPlanRepository.create(subscriptionTier, requestDetails, mockTTL);
+
+          await ethAddressSpendingPlanRepository.save(
+            { ethAddress: aliasAccount.address, planId: hbarSpendingPlan.id },
+            requestDetails,
+            mockTTL,
+          );
+
+          const plan = await ethAddressSpendingPlanRepository.findByAddress(aliasAccount.address, requestDetails);
+          expect(plan.ethAddress).to.eq(aliasAccount.address);
+          expect(plan.planId).to.eq(hbarSpendingPlan.id);
+          const spendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(plan.planId, requestDetails);
+          expect(spendingPlan.active).to.be.true;
+          expect(spendingPlan.amountSpent).to.eq(0);
+          expect(spendingPlan.subscriptionTier).to.eq(subscriptionTier);
+
+          return { aliasAccount, hbarSpendingPlan };
+        };
+
+        const pollForProperAmountSpent = async (
```

Comment:
**Note for reviewers:**
Same idea with pollForProperRemainingHbar() but for hbarSpendingPlans